spec/commands/change_spec.rb

```diff
@@ -0,0 +1,158 @@
+# frozen_string_literal: true
+
+RSpec.describe "bundle change" do
+  before :each do
+    install_gemfile <<-G
+      source "file://#{gem_repo1}"
+
+      gem "rack", "~> 1.0", :group => [:dev]
+      gem "weakling", ">=  0.0.1"
+
+      group :test do
+        gem "rack-test", "= 1.0"
+        gem "rspec"
+      end
+    G
+  end
+
+  describe "when gem is not present" do
+    it "throws error" do
+      bundle "change rake --group dev1"
+
+      expect(out).to include("`rake` could not be found in the Gemfile.")
+    end
+  end
+
+  context "without options" do
+    it "throws error" do
+      bundle "change rack"
+
+      expect(out).to include("Please supply at least one option to change.")
+    end
+  end
+
+  describe "with --group option" do
+    context "when group is present as inline" do
+      it "changes group of the gem" do
+        bundle! "change rack --group dev1"
+
+        gemfile_should_be <<-G
+          source "file://#{gem_repo1}"
+
+          gem "weakling", ">=  0.0.1"
+
+          group :test do
+            gem "rack-test", "= 1.0"
+            gem "rspec"
+          end
+
+          gem "rack", "~> 1.0", :group => [:dev1]
```

Comment:
this should either be `:group => :dev1` or `:groups => [:dev1]`

Reply:
We will have to change the injector https://github.com/bundler/bundler/blob/master/spec/commands/add_spec.rb#L72

Reply:
Handled [here](https://github.com/bundler/bundler/pull/6627)